src/Integration.Vsix/SonarLintTracker/IssuesSnapshot.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Shell.TableControl;
+using Microsoft.VisualStudio.Shell.TableManager;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace SonarLintTracker
+{
+    class IssuesSnapshot : WpfTableEntriesSnapshotBase
+    {
+        private readonly string filePath;
+        private readonly int versionNumber;
+
+        // We're not using an immutable list here but we cannot modify the list in any way once we've published the snapshot.
+        public readonly List<IssueMarker> IssueMarkers = new List<IssueMarker>();
```

Comment:
We don't stick to this practice and it is OK to skip the `ReadOnlyCollection<T>`, but we recently created a new rule about that and we perhaps should start using it.